packages/@okta/vuepress-theme-prose/util/geoLocation.js

```diff
@@ -0,0 +1,78 @@
+import storage from './localStorage'
+
+const STORAGE_KEY = "oktaLocation";
+// Add/Remove country codes from this array to have them blocked/unblocked
+const BLOCKED_REGION_LIST = ['RU', 'BY'];
+// 
+
+class GeoLocation {
+  constructor() {
+    this.hasRetried = false;
+
+    // Add the country code to local storage.
+    this.getLocale();
+  }
+
+  setCountryCode = (countryCode = false) => {
+    if (countryCode) {
+      const date = new Date();
+      const oneWeekInMilliseconds = (1000 * 60 * 60 * 24 * 7);
+      const data = {
+        countryCode: countryCode,
+        expiration: date.getTime() + oneWeekInMilliseconds,
+      };
+      storage.setItem(STORAGE_KEY, JSON.stringify(data));
+    }
+  }
+
+  onSuccess = (response = {}) => {
+    const date = new Date();
+    const storageItem = JSON.parse(storage.getItem(STORAGE_KEY));
+    const expiration = storageItem?.expiration ?? 0;
+    const countryCode = response?.country?.iso_code ?? false;
+    const storageIsExpired = date.getTime() > expiration;
+
+    // Update storage item if it is expired.
+    if (storageIsExpired) {
+      this.setCountryCode(countryCode);
+    }
+
+    // Bail if storage item is set and not expired.
+    if (storageItem !== null && storageIsExpired === false) {
+      return;
+    }
+
+    // Set location storage item.
+    this.setCountryCode(countryCode);
+  }
+
+  getLocale = () => {
+    if (typeof geoip2 !== "undefined") {
+      geoip2.country(this.onSuccess, this.onError);
+    }
+  }
+
+  onError = () => {
+    // Try again if there is an error.
+    if (hasRetried === false) {
+      hasRetried = true;
+      this.getLocale();
+    }
+  }
+}
+
+const isRegionLocked = () => {
+  const storageItem = JSON.parse(storage.getItem(STORAGE_KEY));
```

Comment:
Is there a possible race condition between when `Geolocation` is constructed and when this is read? I know we construct `beforeMount` on the layout, but the API calls might be lagging.